File: Excel/400-499/458/458 Challenge.py
import pandas as pd
import re
import numpy as np

df = pd.read_excel("458 Maximum Consecutive Uppercase Alphabets.xlsx", usecols="A:B")
input = df["Words"].tolist()
test = df["Expected Answer"].tolist()

def get_longest_capital(string):
  caps = re.findall("[A-Z]+", string)
  caps_len = max([len(cap) for cap in caps]) if caps else None
  caps = [cap for cap in caps if len(cap) == caps_len]
  return ", ".join(caps)

output = [get_longest_capital(string) if get_longest_capital(string) else np.nan for string in input]
print(output == test) # True

File: Excel/400-499/458/458 Challenge.R
library(tidyverse)
library(readxl)

input = read_excel("Excel/458 Maximum Consecutive Uppercase Alphabets.xlsx", range = "A1:A11")
test  = read_excel("Excel/458 Maximum Consecutive Uppercase Alphabets.xlsx", range = "B1:B11")

get_longest_capital = function(string) {
  caps = str_extract_all(string, "[A-Z]+") %>% unlist()
  caps_len = ifelse(length(caps) == 0, NA, max(nchar(caps)))
  caps = caps[nchar(caps) == caps_len] %>% paste0(collapse = ", ")
  return(caps)
}

result = input %>%
  mutate(ans = map_chr(Words, get_longest_capital)) %>%
  mutate(ans = ifelse(ans == "", NA_character_, ans))

all.equal(result$ans, test$`Expected Answer`)
# [1] TRUE